### PR TITLE
Fix leaking listeners

### PR DESCRIPTION
**Pull request by damencho**

### CODE REVIEW

File: conference.js [hunk 1]
```diff
@@ -320,25 +319,6 @@ class ConferenceConnector {
...
-            // Schedule reconnect to check if someone else created the room.
-            this.reconnectTimeout = setTimeout(() => {
-                APP.store.dispatch(conferenceWillJoin(room));
-                room.join(null, replaceParticipant);
-            }, 5000);
-
-            const { password }
-                = APP.store.getState()['features/base/conference'];
-
-            AuthHandler.requireAuth(room, password);
```
Comment (saghul) [inline conference.js:337]: Looks like we can get rid of this requireAuth method then.

File: conference.js [hunk 2]
```diff
@@ -320,25 +319,6 @@ class ConferenceConnector {
             break;
         }
 
-        // not enough rights to create conference
-        case JitsiConferenceErrors.AUTHENTICATION_REQUIRED: {
-
-            const replaceParticipant = getReplaceParticipant(APP.store.getState());
-
-            // Schedule reconnect to check if someone else created the room.
```
Comment (saghul) [inline conference.js:328]: Where is this timer now?

File: react/features/etherpad/middleware.js [hunk 3]
```diff
@@ -40,7 +40,7 @@ MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
 StateListenerRegistry.register(
     state => getCurrentConference(state),
     (conference, { dispatch, getState }, previousConference) => {
-        if (conference) {
+        if (conference && !getState()['features/base/conference'].authRequired) {
```
Comment (saghul) [inline react/features/etherpad/middleware.js:43]: Why is this needed? `getCurrentConference` will return the new conference on each willJoin?

File: react/features/follow-me/middleware.js [hunk 4]
```diff
@@ -61,10 +61,12 @@ MiddlewareRegistry.register(store => next => action => {
     case CONFERENCE_WILL_JOIN: {
         const { conference } = action;
 
-        conference.addCommandListener(
-            FOLLOW_ME_COMMAND, ({ attributes }, id) => {
-                _onFollowMeCommand(attributes, id, store);
-            });
+        if (!store.getState()['features/base/conference'].authRequired) {
```
Comment (saghul) [inline react/features/follow-me/middleware.js:64]: Maybe we can add the handler in the JOINED event instead?

**Comment by damencho** [comment]
I think, conference joined can be a bit problematic as it will be racing with some message received events, but CONFERENCE_JOIN_IN_PROGRESS will do the trick. Will work on updating that.

File: react/features/videosipgw/middleware.js [hunk 5]
```diff
@@ -33,7 +33,7 @@ MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
     const result = next(action);
 
     switch (action.type) {
-    case CONFERENCE_WILL_JOIN: {
+    case CONFERENCE_JOIN_IN_PROGRESS: {
         const conference = getState()['features/base/conference'].joining;
```
Comment (saghul) [inline react/features/videosipgw/middleware.js:37]: better take it from the action

**Comment by damencho** [comment]
But I think I broke follow-me and shared-video ...

**Comment by damencho** [comment]
> But I think I broke follow-me and shared-video ...

And its fixed now.

File: modules/UI/authentication/AuthHandler.js [hunk 6]
```diff
@@ -6,7 +6,10 @@ import { openConnection } from '../../../connection';
 import {
     openAuthDialog,
     openLoginDialog } from '../../../react/features/authentication/actions.web';
-import { WaitForOwnerDialog } from '../../../react/features/authentication/components';
+import {
+    LoginDialog as LoginDialogComponents,
```
[the remark this answers is not in the document]
Comment (damencho) [reply modules/UI/authentication/AuthHandler.js]: Done.

File: react/features/base/conference/actionTypes.js [hunk 7]
```diff
@@ -32,6 +32,16 @@ export const CONFERENCE_FAILED = 'CONFERENCE_FAILED';
  */
 export const CONFERENCE_JOINED = 'CONFERENCE_JOINED';
 
+/**
+ * The type of (redux) action which signals that a specific conference joining is in progress.
```
[the remark this answers is not in the document]
Comment (damencho) [reply react/features/base/conference/actionTypes.js:36]: Done.

File: react/features/shared-video/middleware.web.js [hunk 8]
```diff
@@ -1,38 +1,36 @@
...
-        if (conference && conference !== previousConference) {
-            conference.addCommandListener(SHARED_VIDEO,
-                ({ attributes }) => {
-                    const { dispatch, getState } = store;
-                    const { from } = attributes;
-                    const localParticipantId = getLocalParticipant(getState()).id;
-                    const status = attributes.state;
+MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
+    const state = getState();
+    const conference = getCurrentConference(state);
```
[the remark this answers is not in the document]
Comment (damencho) [reply react/features/shared-video/middleware.web.js]: Done.

File: react/features/videosipgw/middleware.js (hunk 5 above)
Comment (damencho) [reply react/features/videosipgw/middleware.js]: Done.

File: react/features/etherpad/middleware.js (hunk 3 above)
Comment (saghul) [reply react/features/etherpad/middleware.js:43]: Are you not refactoring this one?

File: react/features/shared-video/middleware.any.js [hunk 9]
```diff
@@ -36,6 +37,25 @@ MiddlewareRegistry.register(store => next => action => {
     const { ownerId: stateOwnerId, videoUrl: statevideoUrl } = state['features/shared-video'];
 
     switch (action.type) {
+    case CONFERENCE_JOIN_IN_PROGRESS: {
+        conference.addCommandListener(SHARED_VIDEO,
```
Comment (saghul) [inline react/features/shared-video/middleware.any.js:41]: this variable is set from the state, maybe take it from the action? also, getting all those state variables outside of the `case` seems wrong, since it will be done for *any* action! Can you move them inside the case pl?

Comment (saghul) [reply react/features/shared-video/middleware.any.js:41]: Actually, this is super weird. We should be getting those inside the handler, where they are used!

File: react/features/shared-video/middleware.web.js [hunk 10]
```diff
@@ -1,38 +1,37 @@
...
-                    if (status === 'playing') {
-                        if (localParticipantId !== from) {
-                            dispatch(setDisableButton(true));
-                        }
-                    } else if (status === 'stop') {
-                        dispatch(setDisableButton(false));
-                    }
+MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
+    const state = getState();
+    const localParticipantId = getLocalParticipant(state)?.id;
```
Comment (saghul) [inline react/features/shared-video/middleware.web.js:14]: Ditto.

File: react/features/etherpad/middleware.js (hunk 3 above)
Comment (saghul) [reply react/features/etherpad/middleware.js:43]: This is the only thing missing I think!

**Review by saghul — DISMISSED** [review]
LGTM with 1 comment.

**Comment by saghul** [comment]
Looks like you need to rebase...

File: react/features/etherpad/middleware.js (hunk 3 above)
Comment (damencho) [reply react/features/etherpad/middleware.js:43]: Oh I missed that comment, working on it.

**Review by saghul — APPROVED** [review]

**Comment by damencho** [comment]
Chrome tests pass, FF has some problems with the driver.

**Comment by AdalbertoPM** [comment]
Goodnight,
I'm having the same problem, but I don't understand the solution.
Could you be more specific please.
Which files should I modify?

Thanks
Adalberto
